sklearn/externals/joblib/func_inspect.py

```diff
@@ -169,7 +170,11 @@ def getfullargspec(func):
     try:
         return inspect.getfullargspec(func)
     except AttributeError:
-        arg_spec = inspect.getargspec(func)
+        if 3 > sys.version_info[0]:
+           arg_spec = inspect.getargspec(func)
+        else:
+           # getargspec deprecated in python3
+           arg_spec = inspect.signature(func)
```

Comment:
@amueller Digging deeper after questions in joblib I found a previous [PR](https://github.com/scikit-learn/scikit-learn/pull/5301) that may point to a preferred solution.  Will look into this.